src/framed/std/io.clj

```diff
@@ -10,6 +11,11 @@
   "The system temporary directory"
   (System/getProperty "java.io.tmpdir"))
 
+(defn path-join
+  "Join Strings into a String path using the system file separator"
+  [part & parts]
+  (str (Paths/get part (into-array String parts))))
```

Comment:
I don't mind separating out functions. This is used for filesystem operations (_in addition to_ storage, which may need some sorting out as you've observed)